iconservice/icon_service_engine.py

```diff
@@ -1352,20 +1352,30 @@ def _handle_score_invoke(self,
             self._deposit_handler.handle_deposit_request(context, data)
             return None
         else:
-            # charge step
-            if data_type == 'message':
-                # for mainnet backward compatibility
-                if context.revision < Revision.DO_NOT_CHARGE_CONTRACT_CALL_STEP_TO_MESSAGE_DATATYPE.value:
-                    context.step_counter.apply_step(StepType.CONTRACT_CALL, 1)
-            else:
-                # do not charge CONTRACT_CALL step to system SCORE call
-                if to != SYSTEM_SCORE_ADDRESS:
-                    context.step_counter.apply_step(StepType.CONTRACT_CALL, 1)
-
-            # invoke SCORE external method
+            if self._check_contract_call_step(context, to, data_type):
+                context.step_counter.apply_step(StepType.CONTRACT_CALL, 1)
             IconScoreEngine.invoke(context, to, data_type, data)
             return None
 
+    @staticmethod
+    def _check_contract_call_step(context: 'IconScoreContext',
+                                  to: 'Address',
+                                  data_type: str) -> bool:
```

Comment:
'data_type' could be None. To reduce the confusion, need to modify type hint.

Reply:
Agree. I will change `str` to `Optional[str]`